examples/counter/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+use rock_n_rollup::{
+    core::{Application, Runtime},
+    plugins::{
+        database::{Database, Json},
+        logger::Logger,
+    },
+    services::external::{External, FromExternal},
+};
+
+pub struct Decrement {}
+
+impl FromExternal for Decrement {
+    fn from_external(input: Vec<u8>) -> Result<Self, ()> {
+        match input.as_slice() {
+            [0x00] => Ok(Decrement {}),
+            _ => Err(()),
+        }
+    }
+}
+
+pub fn descrement<R: Runtime>(rt: &mut R, _: External<Decrement>) {
+    rt.log("descrement counter");
+}
+
+pub fn counter<L: Logger + Database<Json>>(rt: &mut L) {
+    let counter = 33;
```

Comment:
It would be nice to first get the counter from the storage, if not present you can return the value you want, so that the counter will be persisted between two transition

Reply:
I change it to read from storage at this [commit](4b736d4ef97a9a154bff9ca59eca465d6e0a0833)